### ISSUE TITLE
vxlan interface creation failed for subnet failed to set link up: address already in use

### ISSUE THREAD

**Issue by vvb** [docker/docker#22424]
Noticed the below error.. which creating a container with docker overlay. Happened no matter what subnet I tried with. 

Happened after a docker_overlay network was previously used and had about 1000 containers that were stopped,removed and then docker_overlay network was removed. Recreating it, running the below, showed the issue. 

```
[admin@ucsb-blade2 compose]$ docker network create --driver overlay --subnet=192.169.25.0/16 docker_overlay
05ba96d71a51e7ab3e17eda886036fa9be45907c5035913d082e1b326d3e9081
[admin@ucsb-blade2 compose]$
[admin@ucsb-blade2 compose]$ docker run -d --name=manual --net=docker_overlay jainvipin/web
6f9f0f0ca5b6246ef620135ad48a3e5c071ccc141b8bf25737c329d1fffecf6d
docker: Error response from daemon: 500 Internal Server Error: subnet sandbox join failed for "192.169.0.0/16": vxlan interface creation failed for subnet "192.169.0.0/16": failed to set link up: address already in use.
[admin@ucsb-blade2 compose]$
```
```
[admin@ucsb-blade2 compose]$
[admin@ucsb-blade2 compose]$ docker network create --driver overlay --subnet=192.165.25.0/16 d_overlay
d5470701a45abbe345afd8bed9633f275536c459f4d12e47c53ae3b42d0597f8
[admin@ucsb-blade2 compose]$ docker run -d --name=manual1 --net=d_overlay jainvipin/web
9335a2cf9d466b5d138d80ad340da815f0fd408bcc505b72fd7f59e33e72d77c
docker: Error response from daemon: 500 Internal Server Error: subnet sandbox join failed for "192.165.0.0/16": vxlan interface creation failed for subnet "192.165.0.0/16": failed to set link up: address already in use.
[admin@ucsb-blade2 compose]$
```
```
$ docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:39:25 2016
 OS/Arch:      linux/amd64

Server:
 Version:      swarm/1.1.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   f947993
 Built:        Thu Feb 18 08:28:32 UTC 2016
 OS/Arch:      linux/amd64
```
```
Linux blade2 3.10.0-327.13.1.el7.x86_64 #1 SMP Thu Mar 31 16:04:38 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**UPDATE1**:
the node was rebooted. and the above was tried again. same issue again.

```
[admin@ucsb-blade2 ~]$ docker network ls
NETWORK ID          NAME                          DRIVER
a1eb2d61dec5        ucsb-blade3/bridge            bridge
6d22e22c3014        ucsb-blade5/docker_gwbridge   bridge
e5b68142d589        ucsb-blade5/host              host
36a7c351175d        ucsb-blade2/docker_gwbridge   bridge
9c04d77c371b        ucsb-blade2/host              host
c2e5e6125699        ucsb-blade5/none              null
aa84b99db42e        ucsb-blade2/none              null
4ad72380e2d1        ucsb-blade4/none              null
4d5072608031        ucsb-blade3/host              host
6174a19c0982        ucsb-blade4/docker_gwbridge   bridge
89ac908d3706        ucsb-blade4/host              host
b9a0ea8d6538        ucsb-blade5/bridge            bridge
79fb73b007a7        ucsb-blade3/docker_gwbridge   bridge
cc4c0ff94602        ucsb-blade3/none              null
1a5bef10b888        ucsb-blade2/bridge            bridge
7470cc35c1e3        ucsb-blade4/bridge            bridge
[admin@ucsb-blade2 ~]$
[admin@ucsb-blade2 ~]$
[admin@ucsb-blade2 ~]$ docker network create --driver overlay --subnet=10.5.1.0/16 docker_overlay
48b981161ac929e1bfa5d9941e390fa6ab524a1ba0014e78233e612f2d421c5d
[admin@ucsb-blade2 ~]$
[admin@ucsb-blade2 ~]$ docker run -d --name=m1 --net=docker_overlay jainvipin/web
09171a4d65bb113481dd4ac2cce579788d3bd8b315da9a3131c1f438e2eaf31f
docker: Error response from daemon: 500 Internal Server Error: subnet sandbox join failed for "10.5.0.0/16": vxlan interface creation failed for subnet "10.5.0.0/16": failed to set link up: address already in use.
[admin@ucsb-blade2 ~]$
```

```
[admin@ucsb-blade2 ~]$ docker network inspect docker_overlay
[
    {
        "Name": "docker_overlay",
        "Id": "48b981161ac929e1bfa5d9941e390fa6ab524a1ba0014e78233e612f2d421c5d",
        "Scope": "global",
        "Driver": "overlay",
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.5.1.0/16"
                }
            ]
        },
        "Containers": {},
        "Options": {}
    }
]
[admin@ucsb-blade2 ~]$
```

**UPDATE2**
Haven't been able to recover from this yet. 
Tried removing `/var/lib/docker` `/var/run/docker/netns/`
Uninstalled docker and re-installed it
Created new overlay network. 
same error even after all that

**UPDATE3**
I guess I need to do similar cleanup on the other node in the swarm cluster. After the cleanup on the first node. It works, when it is the only node in the cluster. And starts failing when the second node is added to the swarm cluster.

**UPDATE4**
No amount of clean up helps. Problem shows up when using more than one node in the cluster.
Single node swarm cluster works fine.

**Comment by dvkn** [comment]
I encountered the same error irrespective of the subnet address.  I have a swarm cluster with 5 nodes.

docker run -itd --name=overlay-c1 --net=overlay-net alpine /bin/sh
de4bb19d8839e74b81f433a80e37022625ffc5233a50394fb28b9d0fc682ff39
docker: Error response from daemon: subnet sandbox join failed for "10.1.2.0/24": vxlan interface creation failed for subnet

**Comment by Johnxiaoyi** [comment]
I've encountered the same problem, any progress on the issue?
ERROR: for xxxapp  500 Internal Server Error: subnet sandbox join failed for "172.30.0.0/16": vxlan interface creation failed for subnet "172.30.0.0/16": failed to set link up: address already in use.

I'm pretty sure the 172.30.0.0/16 is  not used in my swarm cluster. btw, the docker version is:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64

**Comment by aboch** [comment]
https://github.com/docker/libnetwork/pull/1215 should fix this issue. docker 1.12 will include this change.

**Comment by cpuguy83** [comment]
@aboch Is this already vendored in?

**Comment by aboch** [comment]
@cpuguy83 yes, it was brought in by #23361

**Comment by cpuguy83** [comment]
Thanks, I'll close this one!

If anyone here wants to test that the issue is fixed, please check out the 1.12 RC from test.docker.com!
